src/Service/QueryParametersParser.php

```diff
@@ -17,23 +19,48 @@ public function parse(array $queryParameters): FiltersInterface
     {
         return Seq::init(function () use ($queryParameters) {
             foreach ($queryParameters as $column => $values) {
-                $values = is_array($values)
-                    ? $values
-                    : ['eq' => $values];
+                $columns = $this->parseColumns($column);
+                $columnsCount = count($columns);
 
-                foreach ($values as $filter => $value) {
-                    yield Tuple::of($column, $filter, new Value($value));
+                foreach ($this->normalizeFilters($values) as $filter => $value) {
+                    Assertion::false($columnsCount > 1 && $filter === 'in', 'Tuples are not allowed in IN filter.');
```

Comment:
What about moving the assertion to a private method more with readable name - eg. `assertTupleIsAllowed()`?